components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -5101,12 +5104,16 @@ private int addSubscription(Connection connection, ApiTypeWrapper apiTypeWrapper
         String apiUUID;
         Identifier identifier;
         String tier;
+        String apiVersion;
+        String majorVersionRange;
 
         //Query to check if this subscription already exists
         String checkDuplicateQuery = SQLConstants.CHECK_EXISTING_SUBSCRIPTION_API_SQL;
         if (!isProduct) {
             identifier = apiTypeWrapper.getApi().getId();
             apiUUID = apiTypeWrapper.getApi().getUuid();
+            String[] contextParts = apiTypeWrapper.getApi().getContext().split("/");
+            apiVersion = contextParts[contextParts.length - 1];
```

Comment:
Why do we get version from the context? can't we take it from the API object?

Reply:
We have access to the API object which is inside of the `apiTypeWrapper`. It doesn't have the `version`. That's why this is taken from the context.
Ref: https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java

Reply:
API version is taken from the `identifier`. ignore the above comment